Source/Serilog.Enrichers.Span/ActivityEnricher.cs

```diff
@@ -40,48 +53,81 @@ public void Enrich(LogEvent logEvent, ILogEventPropertyFactory propertyFactory)
         if (activity is not null)
         {
 #if NET5_0_OR_GREATER
-            AddSpanId(logEvent, activity);
-            AddTraceId(logEvent, activity);
-            AddParentId(logEvent, activity);
+            this.AddSpanId(logEvent, activity);
+            this.AddTraceId(logEvent, activity);
+            this.AddParentId(logEvent, activity);
 #else
-            logEvent.AddPropertyIfAbsent(new LogEventProperty(SpanId, new ScalarValue(activity.GetSpanId())));
-            logEvent.AddPropertyIfAbsent(new LogEventProperty(TraceId, new ScalarValue(activity.GetTraceId())));
-            logEvent.AddPropertyIfAbsent(new LogEventProperty(ParentId, new ScalarValue(activity.GetParentId())));
+            logEvent.AddPropertyIfAbsent(new LogEventProperty(
+                this.propertyNames.SpanId,
+                new ScalarValue(activity.GetSpanId())));
+            logEvent.AddPropertyIfAbsent(new LogEventProperty(
+                this.propertyNames.TraceId,
+                new ScalarValue(activity.GetTraceId())));
+            logEvent.AddPropertyIfAbsent(new LogEventProperty(
+                this.propertyNames.ParentId,
+                new ScalarValue(activity.GetParentId())));
 #endif
         }
     }
-#if NET5_0_OR_GREATER
 
-    private static void AddSpanId(LogEvent logEvent, Activity activity)
+    private static void CheckPropertiesNamesArgument(SpanLogEventPropertiesNames logEventPropertyNames)
+    {
+#if NET6_0_OR_GREATER
+        ArgumentNullException.ThrowIfNull(logEventPropertyNames);
+#else
+        if (logEventPropertyNames is null)
+        {
+            throw new ArgumentNullException(nameof(logEventPropertyNames));
+        }
+#endif
+        if (string.IsNullOrWhiteSpace(logEventPropertyNames.ParentId))
+        {
+            throw new ArgumentException($"the property {nameof(logEventPropertyNames.ParentId)} must not be empty");
```

Comment:
Need to pass the `paramName` argument to the exception. Also, could this validation be moved to the property setter itself? That way, you'd get an exception sooner.

We'd need to keep the null check for `logEventPropertyNames` though.